go/teams/loader.go

```diff
@@ -536,6 +560,50 @@ func (l *TeamLoader) load2InnerLockedRetry(ctx context.Context, arg load2ArgT) (
 	}
 	l.G().Log.CDebugf(ctx, "fullVerifyCutoff: %v", fullVerifyCutoff)
 
+	tracer.Stage("userPreload enable:%v parallel:%v wait:%v",
+		teamEnv.UserPreloadEnable, teamEnv.UserPreloadParallel, teamEnv.UserPreloadWait)
+	preloadCtx, preloadCancel := context.WithCancel(ctx)
+	defer preloadCancel()
+	if teamEnv.UserPreloadEnable {
+		uidSet := make(map[keybase1.UID]struct{})
+		for _, link := range links {
+			// fullVerify definition copied from verifyLink
+			fullVerify := (link.LinkType() != libkb.SigchainV2TypeTeamLeave) ||
+				(link.Seqno() >= fullVerifyCutoff) ||
+				(link.source.EldestSeqno == 0)
+			if !link.isStubbed() && fullVerify {
+				uidSet[link.inner.Body.Key.UID] = struct{}{}
+			}
+		}
+		l.G().Log.CDebugf(ctx, "TeamLoader userPreload uids: %v", len(uidSet))
+		if teamEnv.UserPreloadParallel {
+			// Note this is full-parallel. Probably want pipelining if this is to be turned on by default.
+			var wg sync.WaitGroup
+			for uid := range uidSet {
+				wg.Add(1)
+				go func(uid keybase1.UID) {
+					_, _, err = l.G().GetUPAKLoader().LoadV2(
+						libkb.NewLoadUserArg(l.G()).WithUID(uid).WithPublicKeyOptional().WithNetContext(preloadCtx))
+					if err != nil {
+						l.G().Log.CDebugf(ctx, "error preloading uid %v", uid)
+					}
+					wg.Done()
+				}(uid)
+			}
+			if teamEnv.UserPreloadWait {
+				wg.Wait()
+			}
+		} else {
+			for uid := range uidSet {
+				_, _, err = l.G().GetUPAKLoader().LoadV2(
+					libkb.NewLoadUserArg(l.G()).WithUID(uid).WithPublicKeyOptional().WithNetContext(preloadCtx))
+				if err != nil {
+					l.G().Log.CDebugf(ctx, "error preloading uid %v", uid)
+				}
+			}
+		}
+	}
```

Comment:
i'm curious if we still incur the poll to the server in `verifyLink` to see if the prefetched UPAKs are still warm enough.